examples/aggregator/client/android/java/com/google/oak/aggregator/MainActivity.java

```diff
@@ -0,0 +1,120 @@
+    System.loadLibrary("client_app");
+  }
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    
+    Log.v("Oak", "Start application");
+    setContentView(R.layout.activity_main);
+    
+    Button submitButton = findViewById(R.id.submitButton);
+    submitButton.setOnClickListener(v -> onClick());
+
+    // Set default values.
+    EditText bucketInput = findViewById(R.id.bucketInput);
+    bucketInput.setText("android-test");
+
+    EditText firstKeyInput = findViewById(R.id.firstKeyInput);
+    firstKeyInput.setText("1");
+    EditText firstValueInput = findViewById(R.id.firstValueInput);
+    firstValueInput.setText("10");
+
+    EditText secondKeyInput = findViewById(R.id.secondKeyInput);
+    secondKeyInput.setText("2");
+    EditText secondValueInput = findViewById(R.id.secondValueInput);
+    secondValueInput.setText("20");
+
+    EditText thirdKeyInput = findViewById(R.id.thirdKeyInput);
+    thirdKeyInput.setText("3");
+    EditText thirdValueInput = findViewById(R.id.thirdValueInput);
+    thirdValueInput.setText("30");
+
+    // Set default address (static IP of the Aggregator in Google Cloud).
+    // https://developer.android.com/studio/run/emulator-networking
+    EditText addressInput = findViewById(R.id.addressInput);
+    addressInput.setText("35.246.87.178:8080");
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+  }
+
+  public void onClick() {
+    EditText addressInput = findViewById(R.id.addressInput);
+    String address = addressInput.getText().toString();
+
+    if (!address.equals(rpcAddress)) {
+      Log.v("Oak", "Create channel to: " + address);
+      createChannel(address);
+      rpcAddress = address;
+    }
+
+    Log.v("Oak", "Submit Sample");
+  
+    EditText bucketInput = findViewById(R.id.bucketInput);
+    String bucket = bucketInput.getText().toString();
+
+    ArrayList<Integer> indices = new ArrayList<Integer>();
+    ArrayList<Float> values = new ArrayList<Float>();
+
+    EditText firstKeyInput = findViewById(R.id.firstKeyInput);
+    indices.add(Integer.parseInt(firstKeyInput.getText().toString()));
+    EditText firstValueInput = findViewById(R.id.firstValueInput);
+    values.add(Float.parseFloat(firstValueInput.getText().toString()));
+
+    EditText secondKeyInput = findViewById(R.id.secondKeyInput);
+    indices.add(Integer.parseInt(secondKeyInput.getText().toString()));
+    EditText secondValueInput = findViewById(R.id.secondValueInput);
+    values.add(Float.parseFloat(secondValueInput.getText().toString()));
+
+    EditText thirdKeyInput = findViewById(R.id.thirdKeyInput);
+    indices.add(Integer.parseInt(thirdKeyInput.getText().toString()));
+    EditText thirdValueInput = findViewById(R.id.thirdValueInput);
+    values.add(Float.parseFloat(thirdValueInput.getText().toString()));
+
+    submitSample(bucket, indices, values);
+  }
+
+  private native void createChannel(String address);
```

Comment:
Perhaps add a todo comment to replace the JNI code with java code once TLS is enabled?

Reply:
Done
Created an issue: https://github.com/project-oak/oak/issues/717